iocore/net/UnixNet.cc

```diff
@@ -202,41 +202,28 @@ net_signal_hook_callback(EThread *thread)
 #endif
 }
 
-static void
-net_signal_hook_function(EThread *thread)
-{
-#if HAVE_EVENTFD
-  uint64_t counter = 1;
-  ATS_UNUSED_RETURN(write(thread->evfd, &counter, sizeof(uint64_t)));
-#elif TS_USE_PORT
-  PollDescriptor *pd = get_PollDescriptor(thread);
-  ATS_UNUSED_RETURN(port_send(pd->port_fd, 0, thread->ep));
-#else
-  char dummy = 1;
-  ATS_UNUSED_RETURN(write(thread->evpipe[1], &dummy, 1));
-#endif
-}
-
 void
 initialize_thread_for_net(EThread *thread)
 {
-  new ((ink_dummy_for_new *)get_NetHandler(thread)) NetHandler();
-  new ((ink_dummy_for_new *)get_PollCont(thread)) PollCont(thread->mutex, get_NetHandler(thread));
-  get_NetHandler(thread)->mutex = new_ProxyMutex();
-  PollCont *pc                  = get_PollCont(thread);
-  PollDescriptor *pd            = pc->pollDescriptor;
+  NetHandler *nh = get_NetHandler(thread);
+
+  new ((ink_dummy_for_new *)nh) NetHandler();
+  new ((ink_dummy_for_new *)get_PollCont(thread)) PollCont(thread->mutex, nh);
+  nh->mutex  = new_ProxyMutex();
+  nh->thread = thread;
 
-  thread->schedule_imm(get_NetHandler(thread));
```

Comment:
It breaks the original design that the `NetHandler::mainNetEvent()` as a sub-loop of `EThread::execute()`.
I think we should consider the `UDPNetHandler` and `NetHandler` together. The `PollCont` is used by them for polling and it calls `epoll_wait` to block the `EThread` for awhile.

Reply:
No, that is done via the `tail_cb` rather than being event based. The net handler sub-loop is implemented by setting the tail callback of the event loop to that sub loop. That is much cleaner and clearer than the event based style and makes the event loop more general (in that it's not hard wired to stall on its own condition variable, but can stall on something else specific to the use).

Reply:
Please consider NetHandler and UDPNetHandler run in one EThread. The `tail_cb` is not a common design but Negative Event is.